knowledge_gpt/main.py

```diff
@@ -27,12 +29,18 @@
 st.set_page_config(page_title="KnowledgeGPT", page_icon="📖", layout="wide")
 st.header("📖KnowledgeGPT")
 
+# Initialising submit as False
+if "submit" not in st.session_state:
+    st.session_state["submit"] = False
+
 # Enable caching for expensive functions
 bootstrap_caching()
 
 sidebar()
 
-openai_api_key = st.session_state.get("OPENAI_API_KEY")
+openai_api_key = st.secrets.get("OPENAI_API_KEY")
```

Comment:
I'm not sure we can do this, as this is changing the original structure of the code. If you look here, there seems to be some logic that uses `session_state` rather than `st.secrets`.

For our email and password though, `st.secrets` is perfect